### PR TITLE
Feat: Add swagger documentation for email reg

### PR DESCRIPTION
**Pull request by omikay**
Pull Request Summary

This pull request adds swagger documentation to user email registration functionality.

Changes:

- The response code for "email already exists" changed to 409.
- Swagger documentation for adding and getting subscribers has been done.
- nodemailer configs were added to the .env.example file for better reference.

### CODE REVIEW

File: src/swagger/subscribe/add.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/**
+ * @swagger
+ * /subscribers:
+ *   post:
+ *     tags:
+ *       - Subscribers
+ *     summary: Add a new subscriber
```
Comment (ceydae) [inline src/swagger/subscribe/add.ts:7]: Adds*

File: src/swagger/subscribe/add.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/**
+ * @swagger
+ * /subscribers:
```
Comment (ceydae) [inline src/swagger/subscribe/add.ts:3]: /subscribe

File: src/swagger/subscribe/add.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ *   post:
+ *     tags:
+ *       - Subscribers
```
Comment (ceydae) [inline src/swagger/subscribe/add.ts:6]: Subscribe

File: src/swagger/subscribe/get.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * @swagger
+ * /subscribers:
```
Comment (ceydae) [inline src/swagger/subscribe/get.ts:3]: /subscribe

File: src/swagger/subscribe/get.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ *   get:
+ *     tags:
+ *       - Subscribers
```
Comment (ceydae) [inline src/swagger/subscribe/get.ts:6]: Subscribe

File: src/swagger/subscribe/get.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ *                 status:
+ *                   type: string
+ *                   example: success
+ *                 message:
+ *                   type: string
+ *                   example: Subscribers found successfully
+ *                 result:
+ *                   type: array
+ *                   items:
+ *                     $ref: '#/components/schemas/Subscriber'
```
Comment (ceydae) [inline src/swagger/subscribe/get.ts:26]: result is shown as "string" which is incorrect. It should show the same type as the return type

File: src/swagger/subscribe/add.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+ *               type: object
+ *               properties:
+ *                 status:
+ *                   type: string
+ *                   example: success
+ *                 message:
+ *                   type: string
+ *                   example: Subscriber saved successfully
+ *                 result:
+ *                   $ref: '#/components/schemas/Subscriber'
```
Comment (ceydae) [inline src/swagger/subscribe/add.ts:39]: result is shown as "string" in the swagger UI which is incorrect

**Review by ceydae — CHANGES_REQUESTED** [review]
Use the `authAdmin` middleware for the GET /subscribe so that only admin can see the list of subscribes, not everyone. 

Also, add data to the swagger file of GET /subscribe so that user can enter the `authorization` header in the UI. Take a look at POST /token that has done the exact same thing.

File: src/swagger/subscribe/add.ts (hunk 7 above)
Comment (omikay) [reply src/swagger/subscribe/add.ts:39]: with the components defined, there shouldn't be any problem.

File: src/swagger/subscribe/get.ts (hunk 6 above)
Comment (omikay) [reply src/swagger/subscribe/get.ts:26]: but the type is an array which is the same as the return type

Comment (omikay) [reply src/swagger/subscribe/get.ts:26]: with components defined above, the problem should be solved.

**Review by ceydae — CHANGES_REQUESTED** [review]
swagger does not work (needs authorization header in place)

**Review by ceydae — APPROVED** [review]

**Pull request merged**